pkg/util/ec2/ec2_test.go

```diff
@@ -296,6 +296,28 @@ func TestGetLocalIPv4(t *testing.T) {
 	assert.Equal(t, []string{ip}, ips)
 }
 
+func TestGetPublicIPv4(t *testing.T) {
+	ip := "10.0.0.2"
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Content-Type", "text/plain")
+		switch r.RequestURI {
+		case "/public-ipv4":
+			io.WriteString(w, ip)
+		default:
+			w.WriteHeader(http.StatusNotFound)
+		}
+	}))
+
+	defer ts.Close()
+	metadataURL = ts.URL
+	config.Datadog.Set("ec2_metadata_timeout", 1000)
+	defer resetPackageVars()
+
+	ip, err := GetPublicIPv4()
+	require.NoError(t, err)
+	assert.Equal(t, ip, ip)
```

Comment:
nit, you're shadowing the variable `ip` here so you're comparing it to itself.

Reply:
good catch